tensorflow_datasets/core/folder_dataset/image_folder.py

```diff
@@ -135,15 +137,21 @@ def download_and_prepare(self, **kwargs):  # -> NoReturn:
 
   def _as_dataset(
       self,
-      split,
-      shuffle_files=False,
-      decoders=None,
+      split: str,
+      shuffle_files: bool = False,
+      decoders: Optional[Dict[str, decode.Decoder]] = None,
       read_config=None) -> tf.data.Dataset:
     """Generate dataset for given split."""
     del read_config  # Unused (automatically created in `DatasetBuilder`)
+
     if decoders:
-      raise NotImplementedError(
-          '`decoders` is not supported with {}'.format(type(self).__name__))
+        unrecognized_features = (set(decoders.keys())
```

Comment:
If decoder is invalid, the error should already be raised in `self.info.features.decode_example(..., decoders=decoders)` no ? 
If the error message isn't explicit enough, we should update `features.decode_example` instead. I think it's best to centralize the place where error is raised so error is consistent across datasets.

Reply:
You are right. Removing this code.